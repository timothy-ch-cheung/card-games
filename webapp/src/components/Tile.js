import React from "react";
import styled from 'styled-components';

const Button = styled.button`
font-size: 2.5rem;
width: 100px;
height: 100px;
background-color: #f8f9fa;
border: 0;
flex: 0 0 32%;       
box-shadow: none;
border-radius: 0px;
padding: 0;
:focus {
    outline: 0;
}
:hover {
    background: #eeeeee;
}
`;

class Tile extends React.Component {

    render() {
        return (
            <Button className="btn-tile"></Button>
        );
    }

}

export default Tile;