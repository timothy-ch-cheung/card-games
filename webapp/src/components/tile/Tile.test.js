import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Tile from "./Tile";
import renderer from "react-test-renderer";

afterEach(cleanup);

test('renders tile', () => {
  const { getByTestId } = render(<Tile/>);
  const tile = getByTestId('tile')
  expect(tile).toBeInTheDocument();
});

test('Matches Tile snapshot', () => {
  const tree = renderer.create(<Tile/>).toJSON();
  expect(tree).toMatchSnapshot()
});
