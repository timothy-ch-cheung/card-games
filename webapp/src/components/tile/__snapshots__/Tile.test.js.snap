// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Tile snapshot 1`] = `
<button
  className="sc-AxjAm cYkxHS btn-tile"
  data-testid="tile"
/>
`;

exports[`matches snapshot 1`] = `
<button
  className="sc-AxjAm cYkxHS btn-tile"
  data-testid="tile"
/>
`;
